tests/ModelTest.php

```diff
@@ -876,13 +876,48 @@ public function testMultipleLevelDotNotation(): void
     public function testGetDirtyDates(): void
     {
         $user = new User();
-        $user->setRawAttributes(['name' => 'John Doe', 'birthday' => new DateTime('19 august 1989')], true);
+        $user->name = 'John Doe';
+        $user->birthday = new DateTime('19 august 1989');
+        $user->syncOriginal();
         $this->assertEmpty($user->getDirty());
 
         $user->birthday = new DateTime('19 august 1989');
         $this->assertEmpty($user->getDirty());
     }
 
+    public function testGetDirty()
```

Comment:
Please add a test using embedded documents and arrays to ensure correct behaviour.